src/mark.js

```diff
@@ -85,7 +85,9 @@ function channelSort(channels, x, y) {
   X[1].domain = () => {
     let domain = rollup(range(XV), I => reduce.reduce(I, YV), i => XV[i]);
     domain = sort(domain, reverse ? descendingGroup : ascendingGroup);
-    if (limit < Infinity) domain = domain.slice(0, limit);
+    if (limit !== undefined) {
+      domain = domain.slice(...typeof limit === "object" ? [limit[0], limit[1]] : [0, limit]);
+    }
```

Comment:
here in fact we could be more lazy, since slice(0, undefined) returns the same array.

I the case of limit being an array, I chose to pass [limit[0], limit[1]] rather than limit, but it may be too superstitious